src/OpenCensus.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,119 @@
+    {
+        private readonly PropertyFetcher startRequestFetcher = new PropertyFetcher("Request");
+        private readonly PropertyFetcher stopResponseFetcher = new PropertyFetcher("Response");
+        private readonly PropertyFetcher stopExceptionFetcher = new PropertyFetcher("Exception");
+        private readonly PropertyFetcher stopRequestStatusFetcher = new PropertyFetcher("RequestTaskStatus");
+
+        public HttpHandlerDiagnosticListener(ITracer tracer, ISampler sampler) : base("HttpHandlerDiagnosticListener", tracer, sampler)
+        {
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            if (!(this.startRequestFetcher.Fetch(payload) is HttpRequestMessage request))
+            {
+                // Debug.WriteLine("request is null");
+                return;
+            }
+
+            // TODO: this needs to be generilized
+            if (request.RequestUri.ToString().Contains("zipkin.azurewebsites.net"))
+            {
+                return;
+            }
+
+            this.Tracer.SpanBuilder("HttpOut").SetSampler(this.Sampler).StartScopedSpan();
+            var span = this.Tracer.CurrentSpan;
+            span.PutClientSpanKindAttribute();
+            span.PutHttpMethodAttribute(request.Method.ToString());
+            span.PutHttpHostAttribute(request.RequestUri.Host);
+            span.PutHttpPathAttribute(request.RequestUri.AbsolutePath);
+            span.PutHttpUrlAttribute(request.RequestUri.ToString());
+        }
+
+        public override void OnStopActivity(Activity activity, object payload)
+        {
+            var span = this.Tracer.CurrentSpan;
+
+            if (!(this.stopResponseFetcher.Fetch(payload) is HttpResponseMessage response))
+            {
+                // response could be null for DNS issues, timeouts, etc...
+                // TODO: how do we make sure we will not close a scope that wasn't opened?
+
+                span.End();
+                return;
+            }
+
+            var requestTaskStatus = this.stopRequestStatusFetcher.Fetch(payload) as TaskStatus?;
+
+            if (requestTaskStatus.HasValue)
+            {
+                if (requestTaskStatus != TaskStatus.RanToCompletion)
+                {
+                    span.PutErrorAttribute();
+
+                    if (requestTaskStatus == TaskStatus.Canceled)
+                    {
+                        span.Status = Status.Cancelled;
+                    }
+                }
+            }
+
+            span.PutHttpStatusCode((int)response.StatusCode, response.ReasonPhrase);
+
+            span.End();
+        }
+
+        public override void OnException(Activity activity, object payload)
+        {
+            if (!(this.stopExceptionFetcher.Fetch(payload) is Exception exc))
+            {
+                // Debug.WriteLine("response is null");
+                return;
+            }
+
+            var span = this.Tracer.CurrentSpan;
```

Comment:
please check for null. even though it should never happen, but there may be some bug that will cause current span to be null.

Reply:
done